store/sqlstore/channel_store.go

```diff
@@ -1840,12 +1840,15 @@ func (s SqlChannelStore) CountPostsSince(channelID string, since int64) (int64,
 // UpdateLastViewedAtPost sets a channel as unread for a user at the time of the post selected and update the MentionCount
 // it returns a channelunread so redux can update the apps easily.
 func (s SqlChannelStore) UpdateLastViewedAtPost(unreadPost *model.Post, userID string, mentionCount int) (*model.ChannelUnreadAt, *model.AppError) {
-
 	unread, appErr := s.CountPostsSince(unreadPost.ChannelId, unreadPost.CreateAt)
 	if appErr != nil {
 		return nil, appErr
 	}
 
+	if mentionCount == -1 {
+		mentionCount = int(unread)
```

Comment:
Why all the read posts are marked as mentions? is this for the case where a user want to be notified for every message in the channel? If so, we are counting the read posts, not the unread ones.
I also just realized the variable name is now wrong (originally `CountPostSince` did count the unread) and misleading